tests/api/test_checkout.py

```diff
@@ -1776,3 +1777,218 @@ def test_is_fully_paid_no_payment(checkout_with_item):
     checkout = checkout_with_item
     is_paid = is_fully_paid(checkout, None)
     assert not is_paid
+
+
+@pytest.fixture
+def update_checkout_meta():
+    return """
+    mutation checkoutUpdateMetadata($token: UUID!, $input: MetaInput!) {
+        checkoutUpdateMetadata(token: $token, input: $input){
+            errors {
+                message
+            }
+        }
+    }
+    """
+
+
+@pytest.fixture
+def update_checkout_private_meta():
+    return """
+    mutation checkoutUpdatePrivateMetadata($id: ID!, $input: MetaInput!) {
+        checkoutUpdatePrivateMetadata(id: $id, input: $input){
+            errors {
+                message
+            }
+        }
+    }
+    """
+
+
+@pytest.fixture
+def checkout_meta_update_variables(checkout):
+    return {
+        "token": checkout.token,
+        "input": {
+            "namespace": "test",
+            "clientName": "client1",
+            "key": "foo",
+            "value": "bar",
+        },
+    }
+
+
+@pytest.fixture
+def checkout_private_meta_update_variables(checkout):
+    checkout_id = graphene.Node.to_global_id("Checkout", checkout.pk)
+    return {
+        "id": checkout_id,
+        "input": {
+            "namespace": "test",
+            "clientName": "client1",
+            "key": "foo",
+            "value": "bar",
+        },
+    }
+
+
+def test_user_without_permission_cannot_update_private_meta(
+    staff_api_client,
+    staff_user,
+    update_checkout_private_meta,
+    checkout_private_meta_update_variables,
+):
+    assert not staff_user.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
+    response = staff_api_client.post_graphql(
+        update_checkout_private_meta, checkout_private_meta_update_variables
+    )
+    assert_no_permission(response)
+
+
+def test_user_does_not_need_permission_to_update_meta(
+    staff_api_client, update_checkout_meta, checkout_meta_update_variables, checkout
+):
+    response = staff_api_client.post_graphql(
```

Comment:
You should use `user_api_client`. This will be more meaningful as checkout meta mutation is available for standard users.